File: easy/217-contains-duplicate.ts
/** Given an integer array return if any value appears twice, false otherwise
 *
 * @param {number[]} nums
 * @returns {boolean}
 */
function containsDuplicate(nums: number[]): boolean {
  const seen = new Set();
  for (let num of nums) {
    if (seen.has(num)) return true;

    seen.add(num);
  }

  return false;
}

// Time: O(n), Space: O(n)

File: python/217-contains-duplicate.py
from typing import List

def contains_duplicate(nums: List[int]) -> bool:
    """
        Given an array of integers, returns True if there are duplicates and
        False otherwise.

        e.g. contains_duplicate([1,2,3,1]) -> True
    """

    seen = set()

    for num in nums:
        if num in seen:
            return True

        seen.add(num)

    return False

if __name__ == "__main__":
    assert contains_duplicate([1, 2, 3, 1]) == True
    assert contains_duplicate([1, 2, 3, 4]) == False
    assert contains_duplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]) == True

    print("Passed")